### ISSUE TITLE
Incorrect validation of fields part of subresource/complextype

### ISSUE THREAD

**Issue by deepanjan90** [perfectsense/gyro#201]
**Describe the bug**
Incorrect validation of fields part of subresource/complextype when the field is removed after create.

**To Reproduce**
Gyro up a resource having a subresource/complex type with fields that are required.
Ex. Gyro up https://github.com/perfectsense/gyro-google-provider/tree/master/examples/storage/bucket.gyro having `lifecycle` with `rule` being a required field. https://github.com/perfectsense/gyro-google-provider/tree/master/src/main/java/gyro/google/storage/BucketLifecycle.java#L41

After create remove https://github.com/perfectsense/gyro-google-provider/tree/master/examples/storage/bucket.gyro#L44-L72, all the rules from the `liefecycle` and Gyro up.

**Expected behavior**
Required field validation error for `rule`

**Actual behavior**
No changes

**Additional context** https://github.com/perfectsense/gyro/blob/master/core/src/main/java/gyro/core/command/AbstractConfigCommand.java#L133
When this line executes, the corresponding filescope has the correct values from the config
 - for fields modified belonging to both resource and subresource/complextype
 - for resource fields that have been removed

But incorrect values for subresource/complex type fields that have been removed, they seem to have the refreshed values from the cloud.

The validation of a resource is executed using this filescope and that is the cause of the issue.

**Comment by beetlebugorg** [comment]
Can you test this again. I believe this was fixed by #203. 

```
07:06 $ gyro up storage/bucket.gyro
↓ Loading plugin: gyro:gyro-google-provider:0.99.1-SNAPSHOT
⟳ Refreshed resources: 2

Looking for changes...

⟳ Update google::bucket bucket-1 (example-one) (change lifecycle)
    ⟳ Update lifecycle (change rule)
        - Delete rule action: type = Delete and condition: is-live = false; age = 7
            - Delete action type = Delete
            - Delete condition is-live = false; age = 7

Are you sure you want to change resources? (y/N)
```

**Comment by deepanjan90** [comment]
Fixed by #203
